recirq/otoc/parallel_xeb.py

```diff
@@ -671,55 +589,94 @@ def _alpha_least_square(probs_exp: np.ndarray, probs_data: np.ndarray
 
 
 def _speckle_purity(probs_data: np.ndarray) -> float:
+    """Compute the speckle purity of a probability distribution"""
     d = 4
     return np.var(probs_data) * d ** 2 * (d + 1) / (d - 1)
 
 
 def _pairwise_xeb_probabilities(
-        all_qubits: List[Tuple[int, int]],
-        num_cycle_range: Sequence[int],
-        measured_bits: List[List[List[np.ndarray]]],
-        scrambling_gates: List[List[np.ndarray]],
-        interaction_sequence: Optional[List[Set[Tuple[
-            Tuple[float, float], Tuple[float, float]]]]] = None,
-) -> Tuple[Dict[Tuple[Tuple[int, int], Tuple[int, int]], List[np.ndarray]],
-           Dict[Tuple[Tuple[int, int], Tuple[int, int]], List[np.ndarray]]]:
+    all_qubits: List[Tuple[int, int]],
+    num_cycle_range: Sequence[int],
+    measured_bits: List[List[List[np.ndarray]]],
+    scrambling_gates: List[List[np.ndarray]],
+    interaction_sequence: Optional[
+        List[Set[Tuple[Tuple[float, float], Tuple[float, float]]]]
+    ] = None,
+) -> Tuple[
+    Dict[Tuple[Tuple[int, int], Tuple[int, int]], List[np.ndarray]],
```

Comment:
use gridqubits instead of tuple tuple int int tuple int int

Reply:
Same as above. The code sometimes doesn't recognize the keys if the results are saved/reloaded while Cirq gets updated in-between.